### PR TITLE
Database dependencies

### PR DESCRIPTION
**Pull request by AndyTWF**
- Built on top of and supersedes #53 
- Moves prenotes and wake categories to the database
- Creates a list of dependencies to download in the database and where to find them
- Cannot be merged until plugin updated to reflect new API routes.

### CODE REVIEW

File: app/Http/Controllers/ControllerPositionController.php [hunk 1]
```diff
@@ -14,4 +14,17 @@ public function getAllControllers() : JsonResponse
     {
         return response()->json(ControllerPosition::all());
     }
+
+    public function getControllerPositionsDependency() : JsonResponse
+    {
+        $positions = [];
+        ControllerPosition::all()->each(function (ControllerPosition $position) use (&$positions) {
```
Comment (AxonC) [inline app/Http/Controllers/ControllerPositionController.php]: Could you use the Laravel collection `mapWithKeys` here? https://laravel.com/docs/master/collections#method-mapwithkeys

File: app/Services/AirfieldService.php [hunk 2]
```diff
@@ -2,28 +2,34 @@
...
-        $airfields = Airfield::all();
+        $airfields = [];
+        Airfield::all()->each(function (Airfield $airfield) use (&$airfields) {
+            $prenotePairings = $airfield->prenotePairings()->select(['destination_airfield_id', 'prenote_id'])->get();
+            $prenoteArray = [];
 
-        $airfieldArray = [];
-        $airfields->each(function (Airfield $airfield) use (&$airfieldArray) {
-            $airfieldArray[] = array_merge(
+            $prenotePairings->each(function (Airfield $airfield) use (&$prenoteArray) {
```
Comment (AxonC) [inline app/Services/AirfieldService.php]: Another candidate for `mapWithKeys`?

File: app/Http/Controllers/ControllerPositionController.php (hunk 1 above)
Comment (AndyTWF) [reply app/Http/Controllers/ControllerPositionController.php]: Has been updated :)

File: app/Console/Commands/GenerateRegionalPressures.php [hunk 3]
```diff
@@ -13,25 +14,27 @@
...
     public function handle(RegionalPressureService $service)
     {
-        if ($service->generateRegionalPressures()) {
-            Log::info('Regional pressure settings updated and cached successfully.');
-            $this->info('Regional pressure settings updated and cached successfully.');
+        $this->info('Generating regional pressure settings');
+        $regionalPressures = $service->generateRegionalPressures();
+        if (!is_null($regionalPressures) && count($regionalPressures) !== 0) {
+            event(new RegionalPressuresUpdatedEvent($regionalPressures));
+            Log::info('Regional pressure settings updated successfully.');
```
[the remark this answers is not in the document]
Comment (AndyTWF) [reply app/Console/Commands/GenerateRegionalPressures.php]: Done!

File: app/Services/AirfieldService.php (hunk 2 above)
Comment (AndyTWF) [reply app/Services/AirfieldService.php]: Don't think we can on this one as it's adding to arrays as it goes :)

**Pull request merged**